Завершаем вёрстку всех компонентов

savoyskiy opened the pull request:
Саш, привет.  С новым годом! =) С проверкой можешь не торопиться.
Модальное окно на index включено по умолчанию, можно отключить раскомментировав класс .modal-hidden на последней строке css-файла.

---
:mortar_board: [Завершаем вёрстку всех компонентов](https://up.htmlacademy.ru/htmlcss-individual/1/user/2062579/tasks/11)

savoyskiy committed:
доработки по кодревью

savoyskiy committed:
добавление слайдера цены + доработки для пиксельперфект

savoyskiy committed:
добавлена стилизация чекбоксов и радиокнопок

savoyskiy committed:
добавлены тени, стилизация полей цены, фоны

добавлены тени на страницу + доработана стилизация полей ввода цены + некоторые доработки фонов на переполнение

savoyskiy committed:
переделал тень + доработки по состояниям

savoyskiy committed:
доработаны состояния чекбоксов и радиокнопок

savoyskiy committed:
доработка состояний слайдера

savoyskiy committed:
добавлено модальное окно

savoyskiy committed:
добавлен поповер в модальное окно

savoyskiy committed:
переименовал поповер в тултип

т.к. это по сути тултип =)

savoyskiy committed:
доработка модального окна на переполнение

savoyskiy committed:
исправление для валидации разметки

savoyskiy committed:
изменение кнопки закрытия модалки

для соответствия стайлгайду

savoyskiy committed:
добавление фавиконки

savoyskiy committed:
корректировки значения font-family

savoyskiy committed:
доработки модального окна

1) изменена работа сообщений о неверной дате
2) тултип реализован через чекбокс, появляется при нажатии

savoyskiy committed:
переделал тултип на ховер

меньше кода, проще реализация

savoyskiy committed:
доработки по доступности элементов модалки

savoyskiy committed:
изменены поля в модалке

после просмотра ретро переделал на type="text" для стилизации по макету

savoyskiy committed:
доработка состояний интерактивных элементов

кнопки, инпуты в состояниях ховер и дизэйблд

savoyskiy committed:
доработки подписей к кнопкам и ссылкам

критерий ALLY-02. Подписи интерактивных элементов

savoyskiy committed:
доработки состояния полей ввода

savoyskiy committed:
доработки после проверки на переполнение

savoyskiy committed:
приведение в соответствие с кодгайдом

savoyskiy committed:
доработки по критериям

фавиконка в пнг + доработка работы формы выбора гостиницы

Caneklis commented on index.html:
```diff
@@ -200,5 +204,68 @@ <h2 class="visually-hidden">Лого htmlacademy</h2>
+
+            <label class="modal-form-label date-in" for="modal-date-in">Дата заезда:</label>
+            <div  class="modal-input-wrapper">
+              <input class="modal-date-input date-in" id="modal-date-in" type="text" placeholder="Укажите дату" value="27 апреля 2020">
```
Для всех инпутов добавь атрибут required чтобы форма пустой не отправлялась.

savoyskiy replied:
done

Caneklis commented on styles/styles.css:
```diff
@@ -1279,3 +1468,426 @@ body {
+  opacity: 0.6;
+}
+
+.modal-input-wrapper::after {
+  position: absolute;
+  top: 16px;
+  right: 22px;
+  width: 16px;
+  height: 15px;
+  background-image: url("../images/calendar-icon.svg");
+  background-position: center;
+  opacity: 0.3;
+  content: "";
+  z-index: 1;
+}
```
Обычно это делают кнопкой, по клику на которую будет открываться календарь. С 99% вероятностью проверяющий укажет на это.

savoyskiy replied:
переделал

Caneklis commented on styles/styles.css:
```diff
@@ -1279,3 +1468,426 @@ body {
+
+.control-input-checkbox:disabled + .control-checkbox-alt,
+.control-input-radio:disabled + .control-radio-alt {
+	opacity: 0.3;
```
Я бы еще добавил pointer-events: none чтобы при наведении не отрабатывали стили для hover

savoyskiy replied:
done

Caneklis commented on styles/styles.css:
```diff
@@ -46,14 +47,38 @@ body {
 }
 
+/*тень для страницы слева*/
+body::before {
+  position: absolute;
+  content: "";
+  top: 0;
+  left: 40px;
+  height: 100%;
+  width: 30px;
+  box-shadow: 0 0 15px 0 rgba(0, 0, 0, 0.20);
+  z-index: -1;
+}
+/*тень для страницы справа*/
+body::after {
+  position: absolute;
+  content: "";
+  top: 0;
+  right: 40px;
+  height: 100%;
+  width: 30px;
+  box-shadow: 0 0 15px 0 rgba(0, 0, 0, 0.20);
+  z-index: -1;
+}
+
```
Ты можешь сделать просто тень для body

savoyskiy replied:
done =)

Caneklis reviewed (commented):
Оставил пару комментов, но в целом хорошо и почти готово для защиты. Оставлю пока пулреквест открытым, давай в нем поправишь это и после я приму задание.
1. Проверь как [выглядит](https://disk.yandex.ru/i/NwqDJVrdevDHNg) в фаервоксе 
2. При переполнении [ломается сетка ](https://disk.yandex.ru/i/8OK2m-RJpLKlnw)

savoyskiy committed:
доработки по код-ревью

savoyskiy committed:
доработка формы в модалке

savoyskiy committed:
доработки по кодревью

Переделал форму в каталоге на грид, чтобы лучше работала на переполнение

savoyskiy committed:
доработка модального окна

переделал грид-сетку для лучшей работы с переполнением

savoyskiy commented:
Привет. Внес изменения, коммиты запушил, вроде, должны быть видны в пулл реквесте...

savoyskiy committed:
доработки пиксель-перфект + удаление лишнего

savoyskiy committed:
доработки на переполнение

savoyskiy committed:
переделка модалки для переполнения

savoyskiy committed:
доработки списков на переполнение

savoyskiy committed:
доработка хедера на переполнение

вынес кнопку из списка пользовательского меню + ограничил ширины списков меню в хедере

savoyskiy committed:
доработка чекбоксов и радиокнопок

улучшено отображение в файрфокс и удалено лишнее

savoyskiy committed:
доработка секции сортировки на переполнение

savoyskiy committed:
переработал изображения на переполнение

в блоке преимуществ. будут поля вокруг изображений другого размера, зато картинки отобразятся целиком

savoyskiy committed:
переделал изображения в преимуществах

предыдущий вариант был не оч =/

savoyskiy committed:
переделал чекбоксы и радиокнопки

перенес svg в разметку из фонов, теперь работают даже при незагрузке изображений

Caneklis commented on styles/styles.css:
```diff
@@ -1279,3 +1480,452 @@ body {
+
+.modal-date-input:focus {
+  background-color: #f2f2f2;
+  border: 2px solid #000000;
```
Когда ты добавляешь бордер поле немного двигается. Тут стоит добавить бордер по умолчанию прозрачный, но те же 2 пикселя

savoyskiy replied:
поправил

Caneklis commented on styles/styles.css:
```diff
@@ -429,9 +434,10 @@ body {
 
 .advantages-item-edge img {
-  object-fit: cover;
+  align-self: center;
+  object-fit: contain;
```
Проверяющий скорее всего не согласится с таким вариантом переполнения и не зачтет критерий. Оставляю это на твое усмотрение.

savoyskiy replied:
скорее всего, переделаю весь этот грид - там сейчас не получается однозначно хорошего варианта

Caneklis commented on styles/styles.css:
```diff
@@ -429,9 +434,10 @@ body {
+  object-fit: contain;
   max-width: 100%;
-  height: 100%;
+  background-color: #68a2ca;
```
Для чего это?

savoyskiy replied:
фоновый цвет под картинкой, отличный от фона соседнего блока с текстом - если заменить изображение с другим соотношением сторон, то так видны поля вокруг картинки

Caneklis replied:
Попробуй задай картинке display:block, если я правильно понял о каких полях ты говоришь.

savoyskiy replied:
Не, вот про это:
![FastStoneEditor1](https://github.com/htmlacademy-htmlcss/2062579-sedona-1/assets/78842859/57668f78-7c10-4ca9-b29b-b0d9379cd3c8)

Caneklis commented on styles/styles.css:
```diff
@@ -824,10 +864,183 @@ body {
+  display: block;
+}
+
+.control-checkbox:hover {
+  opacity: 0.6;
+}
```
Сюда стоит добавить cursor: pointer, т.к. ты можешь взаимодействовать с элементом (как с ссылками или инпутами по умолчанию).

savoyskiy replied:
добавил

Caneklis commented on index.html:
```diff
@@ -200,5 +202,74 @@ <h2 class="visually-hidden">Лого htmlacademy</h2>
+
+        <section class="modal-content">
+          <h2 class="modal-heading">Поиск гостиницы в Седоне</h2>
+          <form class="modal-search-form" action="https://echo.htmlacademy.ru/" method="get">
```
Тут могут придраться, что эта форма обратной связи, а не поиска и поэтому нужен method="post". Можешь пока оставить.

savoyskiy replied:
ок

Caneklis commented on index.html:
```diff
@@ -200,5 +202,74 @@ <h2 class="visually-hidden">Лого htmlacademy</h2>
+
+            <label class="modal-form-label date-in" for="modal-date-in">Дата заезда:</label>
+            <div  class="modal-input-wrapper-in">
+              <input class="modal-date-input date-in" id="modal-date-in" name="visit-date-in" type="text" placeholder="Укажите дату" value="27 апреля 2020" required>
```
Я бы убрал value и оставил только placeholder, чтобы по умолчанию форма была не заполнена. Для проверки стилей проверяющий может и сам заполнить если что. Оставляю на твое усмотрение.

savoyskiy replied:
done

Caneklis commented on styles/styles.css:
```diff
@@ -381,6 +389,7 @@ body {
 
 .background-images-wrapper img {
   margin: 51px auto;
+  object-fit: contain;
```
Блоку .background-images-wrapper стоит поменять немного фон, а то welcome сливается.

savoyskiy replied:
поменял

Caneklis reviewed (commented):
Поправь бордер у инпута в форме когда поле в фокусе. Остальное на свое усмотрение и защиту. В целом работу можно отправить на финальную защиту если ты готов.
